ha/core/controllers/pcs/node_controller.py

```diff
@@ -242,12 +304,45 @@ def start(self, nodeid: str) -> dict:
             Log.error(f"{nodeid} status is {_node_status}, node may not be started.")
             raise ClusterManagerError(f"Failed to start {nodeid} as found unhandled status {_node_status}")
 
-class PcsHWNodeController(PcsVMNodeController):
+    @controller_error_handler
+    def stop(self, node_id: str, timeout: int= -1, **op_kwargs) -> dict:
```

Comment:
We need to ensure that "node_id" is not same as the current node right ?
Is that done before this stop function is called or should that be handled here ? 
@ajay-paratmandali @ajaysrivas any comments ?

Reply:
Check not needed; its okay if node_id is same as self node